modules/caddyhttp/fileserver/staticfiles.go

```diff
@@ -333,10 +332,21 @@ func (fsrv *FileServer) ServeHTTP(w http.ResponseWriter, r *http.Request, next c
 		}
 	}
 
+	// if a status code override is configured, write the status code
+	// before serving the file
+	if codeStr := fsrv.StatusCode.String(); codeStr != "" {
+		intVal, err := strconv.Atoi(repl.ReplaceAll(codeStr, ""))
+		if err != nil {
+			return caddyhttp.Error(http.StatusInternalServerError, err)
+		}
+		w.WriteHeader(intVal)
```

Comment:
The problem with this, is that ServeContent can add additional header fields, such as Content-Type, Content-Length, and Content-Range... by writing the header now, we forfeit all of those.

Reply:
Are you sure? I was pretty sure you can still add headers after writing the status?

Reply:
I'm sure; once you call WriteHeader, the header is written, as one might expect.

Not that `io.Copy()` is any better in that sense... i.e. this might not be a new problem, but it'd be nice if we could actually fix it.

Reply:
A bit of googling landed me on https://www.howtobuildsoftware.com/index.php/how-do/dQY/go-serving-static-files-with-a-http-500-status which gave me the idea of wrapping the ResponseWriter to intercept the `WriteHeader` call 🎉 🧠